packages/synthetix-main/contracts/interfaces/IAccountModule.sol

```diff
@@ -3,12 +3,20 @@ pragma solidity ^0.8.0;
 
 import "@synthetixio/core-contracts/contracts/interfaces/ISatelliteFactory.sol";
 import "@synthetixio/core-modules/contracts/interfaces/INftModule.sol";
+import "@synthetixio/core-contracts/contracts/utils/SetUtil.sol";
 
 /// @title Module for managing account token (NFT) and accounts, each account is represented by an NFT
 interface IAccountModule is ISatelliteFactory {
+    struct AccountPermission {
+        address target;
+        bytes32[] roles;
+    }
```

Comment:
IMO data structures are better declared in storage contracts, in this case `AccountModuleStorage.sol`.

Reply:
only reason i just added it to the interface is because it's only a data structure for a return type and not something that's actually stored in storage.